groups/infrastructure/data.tf

```diff
@@ -0,0 +1,33 @@
+data "vault_generic_secret" "secrets" {
+  path = "applications/${var.aws_profile}/${var.environment}/${local.stack_name}-stack"
+}
+
+data "aws_acm_certificate" "cert" {
+  domain = local.domain_name
+}
+
+data "aws_route53_zone" "hosted_zone" {
+  name = local.hosted_zone_name
+}
+
+data "aws_vpc" "vpc" {
+  filter {
+    name = "tag:Name"
+    values = [local.vpc_name]
+  }
+}
+
+data "aws_subnets" "private" {
+  filter {
+    name   = "vpc-id"
+    values = [data.aws_vpc.vpc.id]
+  }
+
+  tags = {
+    NetworkType = "private"
+  }
```

Comment:
This filter is insufficient and instead we should configure a second filter that performs a lookup based on a pattern. Such as in the style of `subnet-*`

This pattern would be held in Vault and looked-up accordingly.